### ISSUE TITLE
How to map timestamp to seconds referring to Issue #3627

### ISSUE THREAD

**Issue by shahzebali42** [facebookresearch/fairseq#3925]
#### segment_start + timestep/total_frames * segment_duration:

I am getting correct timestamps but i want to find respecting seconds of the audio.
Can anyone explain this formulae?

##### What do we have to put in these --> segment_start, segment_duration <-- variables to find respective seconds? 
My audio's sample rate=16000 ,  sample width = 2

**Comment by abarcovschi** [comment]
@shahzebali42 did you ever figure out where to get the values for segment_start and segment_duration?

**Comment by abarcovschi** [comment]
I figured out how to map from timesteps to seconds for each non-blank token. The formula I used is the following:<br />
sec_time = frame_num * (audio_len / (num_frames * sample_rate))<br />
where:
- frame_num = the timestep of the symbol, as returned in the 'timesteps' field of Wl2Decoder.decode() outputs.
- audio_len = the number of samples in the loaded audio file corresponding to the transcript (if using batched w2v2 acoustic model inference, will be zero padded to the length of the longest loaded audio file in the batch). 
- num_frames = the number of frames in the emission matrix returned by the w2v2 acoustic model inference for that audio file (if using batched inference, the number of frames for each audio file will be the same as in this case all loaded audio files are padded to the length of the longest audio file in the batch).
- sample_rate = sample rate of loaded audio files (usually 16000 Hz).
